src/image.cpp

```diff
@@ -456,19 +457,20 @@ namespace Exiv2 {
                             io.seek(restore,BasicIo::beg);
                         }
                     } else if ( option == kpsRecursive && tag == 0x83bb /* IPTCNAA */ ) {
-                        if (offset > std::numeric_limits<uint32_t>::max() - count) {
-                            throw Error(kerArithmeticOverflow);
-                        }
-                        if (static_cast<size_t>(offset + count) > io.size()) {
+
+                        if (static_cast<size_t>(Safe::add(count, offset)) > io.size()) {
                             throw Error(kerCorruptedMetadata);
                         }
-                        size_t   restore = io.tell();  // save
-                        io.seek(offset,BasicIo::beg);  // position
-                        byte* bytes=new byte[count] ;  // allocate memory
-                        io.read(bytes,count)        ;  // read
-                        io.seek(restore,BasicIo::beg); // restore
-                        IptcData::printStructure(out,bytes,count,depth);
-                        delete[] bytes;                // free
+
+                        const size_t restore = io.tell();
```

Comment:
I thought about doing that, but unfortunately large parts of `bigtiffimage.cpp` are copy & pasted from `tiffimage.cpp` with some slight modifications (mostly different field sizes). I'd like to remove that duplication at some point, but it's a larger undertaking, which I'd rather not do in this PR. I can open an issue about that though.

Reply:
@D4N  You're quite correct that bigtiffimage.cpp is a cut'n'paste job on tiffimage.cpp.  I encouraged @Kicer86 to go down that road as it's an easy entry to understanding tiff.  The destination is to have only one tiff processor which can support both tiff and bigtiff.

BigTiff is a complicated project because it will involve a thorough understanding of TiffVisitor.  @Kicer86 is busy with work and family and I've asked @piponazo to consider taking this on.  I've put some notes into the file `WORK-IN-PROGRESS` about this.

I think bigtiff is a v0.28 or v1.0 project.  It's by far the most complex change to Exiv2 that's been attempted in the 10+ years in which I have contributed to the project.

Reply:
That's fine @D4N, feel free to create another issue for it. Otherwise I guess we will find out this issue again when dealing with BiggTiff. I am looking forward to have the RC1 released and we can start applying some of the features are holding back (Movement to C++11, Further CMake changes, etc).